crates/ty_python_semantic/src/types/ide_support.rs

```diff
@@ -314,7 +326,7 @@ impl<'db> AllMembers<'db> {
             // member, e.g., `SomeClass.__delattr__` is not a bound
             // method, but `instance_of_SomeClass.__delattr__` is.
             for Member { name, .. } in all_declarations_and_bindings(db, class_body_scope) {
-                let result = parent_instance.member(db, name.as_str());
+                let result = ty.member(db, name.as_str());
```

Comment:
This is a small bugfix. Instead of trying to access the member on instances of the parent class, just access them on the type that we're adding completions for. If we don't do this, the descriptor protocol will be invoked with the wrong instance type (which is why we see the changes in bound methods).

It was also possible to change this for `TypedDict`, because "instances"/inhabitants of `TypedDict`-based classes are not represented by `Type::NominalInstance`, but `Type::TypedDict`.